### PR TITLE
{tools}[gompi/2023a] OpenMC v0.14.0

### PR DESCRIPTION
**Pull request by grpiccoli**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
@grpiccoli I've made PR to your easybuild-easyconfigs fork, with some changes: https://github.com/grpiccoli/easybuild-easyconfigs/pull/1

A quick summary of the changes:
* move `CMake` to `builddependencies` in `OpenMC`
* use a release tarball for `OpenMC`
* avoid using the bundled dependencies (not present in the tarball anyway, since they are submodules)
* instead add missing build and runtime dependencies (creating easyconfigs where necessary)
* enhance the sanity check
* bump `OpenMC` to `foss`, since one of the dependencies (`xtensor`) pulls in blas/lapack

File: easybuild/easyconfigs/o/OpenMC/OpenMC-0.14.0-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+dependencies = [
+    ('HDF5', '1.14.3'),
+    ('pugixml', '1.14'),
+    ('gsl-lite', '0.41.0'),
+    ('xtensor', '0.25.0'),
+    ('fmt', '10.2.1'),
+    ('Catch2', '3.5.3')
```
Comment (Micket) [inline easybuild/easyconfigs/o/OpenMC/OpenMC-0.14.0-foss-2023a.eb:27]:
```suggestion
    ('HDF5', '1.14.0'),
    ('pugixml', '1.14'),
    ('gsl-lite', '0.41.0'),
    ('xtensor', '0.25.0'),
    ('fmt', '10.1.0'),
    ('Catch2', '2.13.9'),
```
please try to use existing easyconfigs, else we get a complete dependency hell here, with ABI breakage between just about every module.

File: easybuild/easyconfigs/g/gsl-lite/gsl-lite-0.41.0-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+##
+# Authors::   Guillermo Piccoli <guillermo.piccoli@hpcnow.com>
+##
+
+easyblock = 'CMakeMake'
+
+name = 'gsl-lite'
+version = '0.41.0'
+
+homepage = 'https://pugixml.org/'
```
Comment (Micket) [inline easybuild/easyconfigs/g/gsl-lite/gsl-lite-0.41.0-foss-2023a.eb:10]: gsl-lite or pugixml?

**Review by Micket — CHANGES_REQUESTED** [review]
I'm not sure what happened here. 

38 new additions, pretty much all duplicates of things that already exist in the toolchain, a lot of them moved to foss level. 

Duplicate versions of dependencies inside a toolchain version is always a very very painful experience. The easyconfigs repo we provide as default should be viewed as a tested ecosystem where the packages all work together. As such, we can only update things used as dependencies in very rare situations, and even then it's always painful. 

I would expect a PR for a new OpenMC thing to only contain very few easyconfigs, at most maybe 3, like OpenMC itself, a version bump for xtensor which does not have a 2023a variant yet, and then maybe gsl-lite (though your gsl-lite seems to actually be pugixml? it's very confusing). 


Definitely no new python, cryptography, cURL, Boost, etc. etc. 

I'm not sure I would recommend updating this PR anymore, rather, starting fresh using your OpenMC-0.14.0-foss-2023a.eb and xtensor (adjusted to using existing easyconfigs)

**Comment by grpiccoli** [comment]
Thank you @Micket, I will open a new PR with the minimum number of packages required.

**Pull request closed without merging**